tensorflow/python/keras/utils/losses_utils.py

```diff
@@ -62,33 +63,37 @@ def remove_squeezable_dimensions(
     Tuple of `labels` and `predictions`, possibly with last dim squeezed.
   """
   with K.name_scope(name or 'remove_squeezable_dimensions'):
-    predictions = ops.convert_to_tensor_v2_with_dispatch(predictions)
-    labels = ops.convert_to_tensor_v2_with_dispatch(labels)
-    predictions_shape = predictions.get_shape()
+    predictions_spec = tf_utils.get_tensor_spec(predictions)
```

Comment:
Suggestion: change this to:

```python
if not isinstance(predictions, tf.RaggedTensor):
  predictions = ops.convert_to_tensor_v2_with_dispatch(predictions)
if not isinstance(labels, tf.RaggedTensor):
  labels = ops.convert_to_tensor_v2_with_dispatch(labels)
predictions_shape = predictions.shape
predictions_rank = predictions_shape.ndims
labels_shape = labels.shape
labels_rank = labels_shape.ndims
```

(and then I believe the code below original line 70 doesn't need to be modified.)

Reason: as this change is written, I would expect the calls to `ops.convert_to_tensor_v2_with_dispatch` (e.g. on new line 77) to fail if their input is a ragged tensor.  I suspect that your new test doesn't reach any of the `convert_to_tensor_v2_with_dispatch` lines you added.

Reply:
Done.